apps/frontend/assets/src/omg-page-api-detail/index.js

```diff
@@ -0,0 +1,326 @@
+          <Tag
+            icon='Option-Horizontal'
+            title='Details'
+          />
+        </div>
+        <DetailRow
+          label='Label'
+          value={
+            <Input
+              noBorder
+              normalPlaceholder='Label'
+              onChange={e => setLabel(e.target.value)}
+              value={label}
+            />
+          }
+        />
+        {keyType === 'admin' && (
+          <DetailRow
+            label='Global Role'
+            value={
+              <Select
+                noBorder
+                normalPlaceholder='Global Role'
+                onSelectItem={role => setGlobalRole(role.key)}
+                value={_.startCase(globalRole)}
+                options={[
+                  { key: 'super_admin', value: 'Super Admin' },
+                  { key: 'admin', value: 'Admin' },
+                  { key: 'viewer', value: 'Viewer' },
+                  { key: 'none', value: 'None' }
+                ]}
+              />
+            }
+          />
+        )}
+        <DetailRow
+          label='Status'
+          value={
+            <>
+              <span style={{ marginRight: '10px' }}>
+                {status ? 'Active' : 'Inactive'}
+              </span>
+              <Switch
+                open={status}
+                onClick={() => setStatus(!status)}
+              />
+            </>
+          }
+        />
+        <div className='button-group'>
+          <Button
+            styleType='ghost'
+            onClick={() => setView('read')}
+            style={{ minWidth: 'initial' }}
+          >
+            <span>Cancel</span>
+          </Button>
+          <Button
+            styleType='primary'
+            onClick={handleSave}
+            style={{ minWidth: 'initial', marginLeft: '10px' }}
+            disabled={!hasChanged()}
+            loading={loading}
+          >
+            <span>Save</span>
+          </Button>
+        </div>
+      </DetailSection>
+    </Content>
+  )
+}
+
+const ReadView = withRouter(({ assignKey, keyDetail, keyType, setView, location: { pathname } }) => {
+  const id = _.get(keyDetail, 'access_key') || _.get(keyDetail, 'key', '-')
+  return (
+    <Content>
+      <DetailSection>
+        <div className='detail-section-header'>
+          <Tag
+            icon='Option-Horizontal'
+            title='Details'
+          />
+          <Button
+            styleType='ghost'
+            size='small'
+            style={{ minWidth: 'initial' }}
+            onClick={() => setView('edit')}
+          >
+            <span>Edit</span>
+          </Button>
+        </div>
+        <DetailRow
+          label='Type'
+          value={keyType === 'admin' ? 'Admin Key' : 'Client Key'}
+        />
+        <DetailRow
+          label='ID'
+          value={<Id maxChar={200}>{id}</Id>}
+        />
+        <DetailRow
+          label='Label'
+          value={<div>{_.get(keyDetail, 'name') || '-'}</div>}
```

Comment:
ok super random between using fallback and not using it 🤣

Reply:
because `null` is seen as a value by `_.get` and so it won't return the fallback `' '`. and i need the empty string to pass into input component or i get `Warning: `value` prop on `input` should not be null. Consider using an empty string to clear the component or `undefined` for uncontrolled components.` error 🤦‍♂